File: netty-loop/src/main/java/com/wenyize/thrift/PersonServiceImpl.java
package com.wenyize.thrift;

import com.wenyize.thrift.generated.DataException;
import com.wenyize.thrift.generated.Person;
import com.wenyize.thrift.generated.PersonService;
import org.apache.thrift.TException;

public class PersonServiceImpl implements PersonService.Iface {
    @Override
    public Person getPersonByUserName(String username) throws DataException, TException {
        System.out.println("Got Client Param: " +username);

        Person person = new Person();

        person.setUsername(username);
        person.setAge(20);
        person.setMarried(false);

        return person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {

        System.out.println("Got Client Param: ");
        System.out.println(person.getUsername());
        System.out.println(person.getAge());
        System.out.println(person.isMarried());
    }
}

File: pythrift/PersonServiceImpl.py
__author__ = "作者"

from py.thrift.generated import ttypes

class PersonServiceImpl:


    def getPersonByUserName(self,username):

        print("got from client param:",username)

        person = ttypes.Person()
        person.username = "python实现"
        person.age = 1
        person.married = False

        return person

    def savePerson(self,person):
        print("Got save call ,param: ")

        print(person.username)
        print(person.age)
        print(person.married)